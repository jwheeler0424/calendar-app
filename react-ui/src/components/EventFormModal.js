import React from 'react';
import Modal from 'react-modal';

const EventFormModal = (props) => {
    
    return (
        <Modal
            isOpen={props.messages.length >= 1 ? true : false}
            onRequestClose={props.handleClearMessages}
            contentLabel="Error!"
            closeTimeoutMS={200}
            className="modal"
            overlayClassName="overlay"
            ariaHideApp={false}
        >
            <h3 className="modal__title">Error!</h3>
            <p className="modal__body">The following errors occured on Event submission:</p>
            <ul className="modal__list">
                {props.messages && props.messages.map((message, index) => (
                    <li key={index}>{message}</li>
                ))}
            </ul>
            <button className="button button--modal" onClick={props.handleClearMessages}>Okay</button>
        </Modal>
    )
};

export { EventFormModal as default };