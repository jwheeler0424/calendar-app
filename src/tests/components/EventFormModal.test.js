import React from 'react';
import { shallow } from 'enzyme';
import EventFormModal from '../../components/EventFormModal';

const messages = ['Please enter a title for the event.']

test('should render EventFormModal correctly', () => {
    const wrapper = shallow(<EventFormModal messages={messages} />);
    expect(wrapper).toMatchSnapshot();
});

test('should close on button click', () => {
    const handleClearMessages = jest.fn();
    const wrapper = shallow(<EventFormModal  messages={messages} handleClearMessages={handleClearMessages} />);
    wrapper.find('button').simulate('click'); 
    expect(handleClearMessages).toHaveBeenCalled();
})