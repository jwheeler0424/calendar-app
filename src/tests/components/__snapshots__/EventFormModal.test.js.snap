// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EventFormModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentElement={[Function]}
  contentLabel="Error!"
  isOpen={true}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Error!
  </h3>
  <p
    className="modal__body"
  >
    The following errors occured on Event submission:
  </p>
  <ul
    className="modal__list"
  >
    <li
      key="0"
    >
      Please enter a title for the event.
    </li>
  </ul>
  <button
    className="button"
  >
    Okay
  </button>
</Modal>
`;
